app/scripts/controllers/transactions/index.js

```diff
@@ -1090,4 +1072,43 @@ export default class TransactionController extends EventEmitter {
       }
     }
   }
+
+  _trackTxEvent(txMeta, event, extraOpts = {}) {
+    const {
+      type,
+      status,
+      chainId,
+      origin: referrer,
+      txParams: { gasPrice },
+      metamaskNetworkId: network,
+    } = txMeta;
+    const source = referrer === 'metamask' ? 'user' : 'dapp';
+
+    this._trackMetaMetricsEvent({
+      event,
+      category: 'Transactions',
+      sensitiveProperties: {
+        type,
+        status,
+        gasPrice,
+        referrer,
+        source,
+        network,
+        chain_id: chainId,
+        ...extraOpts,
+      },
+    });
+  }
+
+  _failTransaction(txId, error) {
+    this.txStateManager.setTxStatusFailed(txId, error);
+    const txMeta = this.txStateManager.getTransaction(txId);
+    this._trackTxEvent(txMeta, 'Transaction Finalized');
```

Comment:
lets go ahead and include the error message in the event extra opts here because this could be an on_chain error and I think that'll be really valuable in assessing how we might better serve our customers.

Reply:
good though! error message added